### PR TITLE
[doc] Update mixed precision

### PR DESCRIPTION
**Pull request by jiangmingyan**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#3764

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

update mixed precision training tutorial 

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: docs/source/en/features/mixed_precision_training.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Auto Mixed Precision Training
+# Auto Mixed Precision Training(antiquated)
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training.md:1]: Please add a space before bracket.

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
+# Auto Mixed Precision Training(latest)
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:1]: Please add a space before bracket.

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+Author: Chuanrui Wang, Shenggui Li, Yongbin Li
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:3]: This new file should be added in sidebars.json

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+| AMP_TYPE.APEX | ❌ | ❌ | More fine-grained, we can choose opt_level O0, O1, O2, O3 |
+| AMP_TYPE.NAIVE | ✅ | ✅ | Model parameters, forward and backward operations are all downcast to fp16 |
+
+The first two rely on the original implementation of PyTorch (version 1.6 and above) and NVIDIA Apex.
+The last method is similar to Apex O2 level.
+Among these methods, apex AMP is not compatible with tensor parallelism.
+This is because that tensors are split across devices in tensor parallelism, thus, it is required to communicate among different processes to check if inf or nan occurs in the whole model weights.
+We modified the torch amp implementation so that it is compatible with tensor parallelism now.
+
+> ❌️ fp16 and zero configuration are not compatible
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:35]: Please remove `configuration`

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+<figure style={{textAlign: "center"}}>
+<img src="https://s2.loli.net/2022/01/28/URzLJ3MPeDQbtck.png"/>
+<figcaption>Illustration of an ordinary AMP (figure from <a href="https://arxiv.org/abs/2108.05818">PatrickStar paper</a>)</figcaption>
+</figure>
+
+## AMP in Colossal-AI
+
+We supported three AMP training methods and allowed the user to train with AMP with no code. If you want to train with amp, just assign `mixed_precision` with `fp16` when you instantiate the `Booster`. Now booster support torch amp, the other two(apex amp, naive amp) are still stared by `colossalai.initiate`, if needed, please refer to [this](./mixed_precision_training.md). Next we will support `bf16`, `fp8`.
+
+### start with booster
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:68]: The first letter should be uppercase.

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+### Naive AMP Configuration
+
+In Naive AMP mode, we achieved mixed precision training while maintaining compatibility with complex tensor and pipeline parallelism.
+This AMP mode will cast all operations into fp16.
+The following code block shows the `config.py` file for this mode.
+
+```python
+from colossalai.mixed_precision import FP16NaiveMixedPrecision
+
+mixed_precision = FP16ApexMixedPrecision(
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:178]: Wrong class

File: docs/sidebars.json [hunk 7]
```diff
@@ -45,6 +45,7 @@
         "features/gradient_clipping",
         "features/gradient_handler",
         "features/zero_with_chunk",
+        "features/mixed_precision_training_with_booster",
```
Comment (ver217) [inline docs/sidebars.json:48]: Move this above the old doc

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+from colossalai.mixed_precision import FP16TorchMixedPrecision
+
+mixed_precision = FP16TorchMixedPrecision(
+    init_scale=2.**16,
+    growth_factor=2.0,
+    backoff_factor=0.5,
+    growth_interval=2000)
+```
+
+With optional arguments:
```
[the remark this answers is not in the document]
Comment (ver217) [reply docs/source/en/features/mixed_precision_training_with_booster.md:97]: You can delete these description after using autodoc.

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+>
+> ⚠️ Pipeline only support naive AMP currently
+
+We recommend you to use torch AMP as it generally gives better accuracy than naive AMP if no pipeline is used.
+
+## Table of Contents
+
+In this tutorial we will cover:
+
+1. AMP introduction
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:45]: You can use a link here, which is more user-friendly, like https://github.com/hpcaitech/ColossalAI/blob/32f81f14d480b579a93fd1786fa38b8c2de79189/docs/source/en/basics/booster_plugins.md?plain=1#L14

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+<figcaption>Illustration of an ordinary AMP (figure from <a href="https://arxiv.org/abs/2108.05818">PatrickStar paper</a>)</figcaption>
+</figure>
+
+## AMP in Colossal-AI
+
+We supported three AMP training methods and allowed the user to train with AMP with no code. If you want to train with amp, just assign `mixed_precision` with `fp16` when you instantiate the `Booster`. Now booster support torch amp, the other two(apex amp, naive amp) are still stared by `colossalai.initialize`, if needed, please refer to [this](./mixed_precision_training.md). Next we will support `bf16`, `fp8`.
+
+### Start with Booster
+instantiate `Booster` with `mixed_precision="fp16"`, then you can train with torch amp.
+```python
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:68]: Should this code be tested? If not, please wrap it with `<!--- doc-test-ignore-start -->`, see https://github.com/hpcaitech/ColossalAI/tree/main/docs#-doc-testing

File: docs/source/zh-Hans/features/mixed_precision_training_with_booster.md [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+        booster.backward(loss, optimizer)
+        optimizer.step()
+    lr_scheduler.step()
+```
+
+### 步骤 6. 启动训练脚本
+
+使用下列命令启动训练脚本，你可以改变 `--nproc_per_node` 以使用不同数量的 GPU。
+
+```python
```
Comment (ver217) [inline docs/source/zh-Hans/features/mixed_precision_training_with_booster.md:268]: This should be `shell` code instead of `python`

File: docs/source/en/features/mixed_precision_training_with_booster.md [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+for epoch in range(NUM_EPOCHS):
+    for img, label in enumerate(train_dataloader):
+        img = img.cuda()
+        label = label.cuda()
+        optimizer.zero_grad()
+        output = model(img)
+        loss = criterion(output, label)
+        booster.backward(loss, optimizer)
+        optimizer.step()
+        lr_scheduler.step()
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training_with_booster.md:241]: Why do we call `step()` twice?

File: docs/source/en/features/mixed_precision_training.md [hunk 13]
```diff
@@ -365,3 +365,4 @@ Use the following command to start the training scripts. You can change `--nproc
 ```python
```
Comment (ver217) [inline docs/source/en/features/mixed_precision_training.md:365]: This is not python code. Please fix this.
